src/sentry/buffer/redis.py

```diff
@@ -208,6 +208,63 @@ def get(
             col: (int(results[i]) if results[i] is not None else 0) for i, col in enumerate(columns)
         }
 
+    def get_redis_connection(self, key: str) -> Any | None:
+        # TODO type RedisPipeline instead of using Any here
+        if is_instance_redis_cluster(self.cluster, self.is_redis_cluster):
+            conn = self.cluster
+        elif is_instance_rb_cluster(self.cluster, self.is_redis_cluster):
+            conn = self.cluster.get_local_client_for_key(key)
+        else:
+            raise AssertionError("unreachable")
+
+        pipe = conn.pipeline()
+        return pipe
+
+    def push_to_set(self, key: str, value: list[int] | int) -> None:
+        pending_key = self._make_pending_key_from_key(key)
+        pipe = self.get_redis_connection(pending_key)
+        if pipe:
+            pipe.sadd(key, value)
+            pipe.expire(key, self.key_expire)
+            pipe.execute()
+
+    def get_set(self, key: str) -> None:
+        pending_key = self._make_pending_key_from_key(key)
+        pipe = self.get_redis_connection(pending_key)
+        if pipe:
+            pipe.smembers(key)
+            return pipe.execute()
```

Comment:
looking at this, push to set, and push to hash there's a lot of common code, i wonder if we could create a helper function that wraps creating the key, getting the connection, validating the pipe, and then triggering execute and returning the result.